### PR TITLE
(CDPE-6808) Add podman support

### PR DESCRIPTION
**Pull request by abottchen**
This commit adds support for podman. Instead of just executing docker, the task will look to see if either podman or docker are present, using the first one found.

An environment variable is provided to give users a way to override the automatic detection if there is a special case we haven't thought of.

Split out the runtime specific tests to separate specs, one for podman and another for docker to allow them to run on github runners with the appropriate runtime available.

To keep this PR a bit more readable, the functional changes are in the first commit.  The second commit is just switching function and variable names to no longer refer to docker when not appropriate.

### CODE REVIEW

File: .github/workflows/unit-test.yml [hunk 1]
```diff
@@ -2,15 +2,38 @@ name: unit-test
...
+          sudo apt-get remove -y docker-ce
+      # podman should already be on the runner, but doesn't seem to be on the PATH
+      # Re-installing fixes this
+      - name: Install podman
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y podman
+      - run: bundle exec rspec spec/podman_spec.rb
   unix_tests:
-    runs-on: macos-latest
```
Comment (abottchen) [inline .github/workflows/unit-test.yml:6]: Not exactly sure why this was ever set to macos, as that doesn't have a runtime installed. Switching over to ubuntu so we at least have one.

File: spec/run_cd4pe_job_spec.rb [hunk 2]
```diff
@@ -308,7 +316,7 @@
       output = job_helper.run_job
 
       expect(output[:job][:exit_code]).to eq(127)
-      expect(output[:job][:message].end_with?("command not found\n")).to be(true)
+      expect(output[:job][:message].end_with?("not found\n")).to be(true)
```
Comment (abottchen) [inline spec/run_cd4pe_job_spec.rb:319]: The file not found error is different on ubuntu vs. mac, so this had to change.

File: spec/run_cd4pe_job_spec.rb [hunk 3]
```diff
@@ -157,75 +157,6 @@
       expect(ENV['REPO_DIR']).to eq("#{@working_dir}/cd4pe_job/repo")
     end
   end
-
```
Comment (abottchen) [inline spec/run_cd4pe_job_spec.rb:160]: I moved all of the container specific tests to separate specs to allow them to be tested on runners with runtimes present.

File: tasks/run_cd4pe_job.json [hunk 4]
```diff
@@ -24,15 +24,15 @@
     },
     "docker_image": {
```
Comment (abottchen) [inline tasks/run_cd4pe_job.json:25]: I'm leaving the parameter names the same even though they have "docker" in them to make this not represent a breaking change.  That said, these parameters are not really used by customers, just by our cd4pe_deployments module when invoking jobs. So changing them and causing the additional cost of forcing customers to align their versions doesn't seem worth it.

**Review by Magisus — APPROVED** [review]
This all makes sense to me. And it doesn't seem terribly complicated, which is nice.

**Pull request merged**